pkg/build/graph/analysis/bc.go

```diff
@@ -162,6 +162,62 @@ func FindCircularBuilds(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	return markers
 }
 
+// findPendingTags is the guts behind FindPendingTags .... break out some of the content and reduce some indentation
+func findPendingTags(uncastIstNode graph.Node, g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+	istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+	if istNode.Found() {
+		return markers
+	}
+	bcNodes := buildedges.BuildConfigsForTag(g, uncastIstNode)
+	for _, bcNode := range bcNodes {
+		latestBuild := buildedges.GetLatestBuild(g, bcNode)
+
+		// A build config points to the non existent tag but no current build exists.
+		if latestBuild == nil {
+			markers = append(markers, osgraph.Marker{
+				Node:         graph.Node(bcNode),
+				RelatedNodes: []graph.Node{uncastIstNode},
+
+				Severity:   osgraph.WarningSeverity,
+				Key:        TagNotAvailableWarning,
+				Message:    fmt.Sprintf("%s needs to be imported or created by a build.", f.ResourceName(istNode)),
```

Comment:
thanks @kargakis - I'll go with something like the snippet in my prior comment.
Also, looking at the use of the `Node` field for `Marker` in bc.go, it does not appear to always be a build config node.  As such, it would be easier to reference the istNode in the `Node` field and the array bcNodes in the `RelatedNodes` field.

I'll start with the change in mind, but let me know how flexible we can be with these fields and I can shift direction if need be.